src/graph/heterograph_capi.cc

```diff
@@ -612,6 +613,10 @@ DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroReverse")
     }
     // node types are not changed
     const auto& num_nodes = g->NumVerticesPerType();
-    *rv = CreateHeteroGraph(hg->meta_graph(), rev_ugs, num_nodes);
+    const auto& meta_edges = hg->meta_graph()->Edges("eid");
+    const auto& rev_meta = ImmutableGraph::CreateFromCOO(
+                                                         hg->meta_graph()->NumVertices(),
+                                                         meta_edges.dst, meta_edges.src);
+    *rv = CreateHeteroGraph(rev_meta, rev_ugs, num_nodes);
```

Comment:
The metagraph does not matter and we will probably delete them later. The correct fix should be:
```c++
const auto& num_nodes = g->NumVerticesPerType();
if (num_nodes.size() == 1)
  *rv = CreateHeteroGraph(hg->meta_graph(), rev_ugs, num_nodes);
else
  *rv = CreateHeteroGraph(hg->meta_graph(), rev_ugs, {num_nodes[1], num_nodes[0]});
```

Reply:
Why metagraph doesn't matter?  The edges in the metagraph should also be reversed, right?

Reply:
I think @jermainewang 's proposal works for bipartite graph, however for a general heterograph with multiple etypes ( > 2) we cannot simply reverse the number of nodes array.